mirascope/core/base/_utils.py

```diff
@@ -181,20 +197,23 @@ def convert_function_to_base_tool(
     model = create_model(
         fn.__name__,
```

Comment:
here, we should use `f"{namespace}.{fn.__name__}" if namespace else fn.__name__`